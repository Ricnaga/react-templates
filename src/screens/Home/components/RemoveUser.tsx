import { Button, ModalFooter } from '@chakra-ui/react';

import { useAxios } from '@application/api/axios/useAxios';
import { useSnackbar } from '@shared/components/Snackbar/useSnackbar';

type RemoveUserUserProps = {
  onClose: () => void;
  id: string;
  onRefetchUsers: () => Promise<void>;
};

export function RemoveUser({
  onClose,
  id,
  onRefetchUsers,
}: RemoveUserUserProps) {
  const { onSnackBar } = useSnackbar();
  const {
    functions: { callDELETE },
  } = useAxios();

  const onDelete = async () =>
    callDELETE<Record<'data', unknown>>({
      url: `/user/${id}`,
    })
      .then(() => onRefetchUsers())
      .then(() => onSnackBar('This user was deleted'))
      .catch(() => onSnackBar('This user was not deleted', 'error'));

  return (
    <ModalFooter>
      <Button colorScheme="blue" mr={3} type="submit" onClick={onDelete}>
        Remover
      </Button>
      <Button variant="ghost" onClick={onClose}>
        Fechar
      </Button>
    </ModalFooter>
  );
}
