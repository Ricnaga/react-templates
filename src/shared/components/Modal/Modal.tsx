import {
  Modal as ChakraModal,
  ModalCloseButton,
  ModalContent,
  ModalHeader,
  ModalOverlay,
} from '@chakra-ui/react';
import { ReactNode } from 'react';

type ModalProps = {
  isOpen: boolean;
  onClose: () => void;
  children: ReactNode;
  title?: string;
};

export function Modal({ children, isOpen, onClose, title }: ModalProps) {
  return (
    <ChakraModal isOpen={isOpen} onClose={onClose} closeOnOverlayClick={false}>
      <ModalOverlay />
      <ModalContent>
        {title && <ModalHeader>{title}</ModalHeader>}
        <ModalCloseButton />
        {children}
      </ModalContent>
    </ChakraModal>
  );
}
