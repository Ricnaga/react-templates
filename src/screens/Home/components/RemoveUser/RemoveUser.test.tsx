import { renderWithTheme } from '@application/test/testing-library';
import { Modal } from '@shared/components/Modal/Modal';

import { RemoveUser } from './RemoveUser';

let containerElement: HTMLElement;

describe('Component: RemoveUser', () => {
  beforeEach(() => {
    const { container } = renderWithTheme(
      <Modal isOpen onClose={jest.fn().mockReturnValue(true)}>
        <RemoveUser onClose={jest.fn} onRefetchUsers={jest.fn()} id="1" />,
      </Modal>,
    );
    containerElement = container;
  });

  it('should render correctly', () => {
    expect(containerElement).toMatchSnapshot();
  });
});
