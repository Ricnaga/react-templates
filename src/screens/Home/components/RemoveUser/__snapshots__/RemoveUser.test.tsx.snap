// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RemoveUser should render correctly 1`] = `
<div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
